src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -158,7 +158,7 @@ public string GeneratePreContent()
         /// </returns>
         public string GenerateContent()
         {
-            if (SelfClosing)
+            if (!string.IsNullOrWhiteSpace(TagName) && SelfClosing)
```

Comment:
@NTaylorMullen thanks. that is a good point. How about _isTagNameNullOrEmpty?? I'll update this code when I do actual PR.